code/game/machinery/reagents/pump.dm

```diff
@@ -0,0 +1,256 @@
+
+/obj/machinery/pump
+	name = "fluid pump"
+	desc = "A fluid pumping machine."
+	anchored = 0
+	density = 1
+
+	icon = 'icons/obj/machines/reagent.dmi'
+	icon_state = "pump"
+
+	circuit = /obj/item/weapon/circuitboard/fluidpump
+
+	var/on = 0
+	var/obj/item/weapon/cell/cell = null
+	var/use = 200
+	var/efficiency = 1
+	var/reagents_per_cycle = 40
+	var/unlocked = 0
+	var/open = 0
+
+	var/obj/item/hose_connector/output/Output
+
+// Overlay cache vars.
+	var/icon/liquid
+	var/icon/tank
+	var/icon/powerlow
+	var/icon/glass
+	var/icon/open_overlay
+	var/icon/cell_overlay
+
+/obj/machinery/pump/Initialize()
+	create_reagents(200)
+	. = ..()
+	default_apply_parts()
+
+	if(ispath(cell))
```

Comment:
This allows subtypes or map-edits to dictate their celltype by inputting a path. They're meant to start empty by default.

Reply:
Might be out of scope, but I think this usage is actually better, and the other cases where its used should be adapted to this. Since it's actually, y'know, customizable